main.go

```diff
@@ -309,11 +312,47 @@ func (c *controller) generateSecrets() []*v1.Secret {
 	return secrets
 }
 
+func userHomeDir() string {
+	if runtime.GOOS == "windows" {
+		home := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
+		if home == "" {
+			home = os.Getenv("USERPROFILE")
+		}
+		return home
+	}
+	return os.Getenv("HOME")
+}
+
+func getDefaultAWSCredentials() (string, string) {
+	awsAccountID, awsRegion := "", ""
+
+	homeDir := userHomeDir()
+	credPath := filepath.Join(homeDir, ".aws/credentials")
+
+	if _, err := os.Stat(credPath); !os.IsNotExist(err) {
+		output, err := exec.Command("aws", "sts", "get-caller-identity", "--output", "text", "--query", "Account").Output()
+		if err != nil {
+			logrus.Error("Get AWS accountID from default credential file error!", err)
```

Comment:
in case of error, should we return it to the caller and let it decide whether to stop the execution (fatal) or not?

Reply:
yes, seems to be better, and I think this PR is not checked for about one year, I will close it :)